query-service-impl/src/main/java/org/hypertrace/core/query/service/prometheus/PromQLMetricResponse.java

```diff
@@ -0,0 +1,97 @@
+package org.hypertrace.core.query.service.prometheus;
+
+import com.fasterxml.jackson.annotation.JsonAlias;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.DeserializationFeature;
+import com.fasterxml.jackson.databind.JsonDeserializer;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Singular;
+
+@Getter
+@Builder
+@NoArgsConstructor
+@AllArgsConstructor
+public class PromQLMetricResponse {
+  private static final ObjectMapper OBJECT_MAPPER =
+      new ObjectMapper().configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+
+  @JsonProperty("status")
+  private String status;
+
+  @JsonProperty("data")
+  private PromQLData data;
+
+  @Getter
+  public static class PromQLData {
+    @JsonProperty("resultType")
+    private String resultType;
+
+    @JsonProperty("result")
+    private List<PromQLMetricResult> result;
+  }
+
+  @Getter
+  @Builder
+  @NoArgsConstructor
+  @AllArgsConstructor
+  @EqualsAndHashCode
+  public static class PromQLMetricResult {
```

Comment:
Pl. let me know, can move them to individual classes. Added here for ease of reading with https://prometheus.io/docs/prometheus/latest/querying/api/#instant-queries